### DISCUSSION TITLE
Trigger selection api endpoint only of first select is done

### DISCUSSION THREAD

**Discussion by gevezex** [vueform/vueform#148] [Questions]
Hi Guys,

I have 2 select elements. selectBrand and apiSelect. I Want the endpoint api of the apiSelect triggered only when one of the options of selectBrand has been selected. As this variable is used by apiSelect. The setup is working but it creates a lot of errors in the console log/error when nothing is selected in selectBrand. I tried the "Conditions" option but there doesn't look anything what I can use (from the vueform demo). 
I want to have both elements visible. But the data retrieval of apiSelect should be triggered only when one of the options of  "selectBrand" is selected.

I have an example on https://4v85cl-5173.csb.app/

Any help is appreciated.

**Comment by nclemeur** [comment]
Looks like you link is not reachable: "Unauthorized Preview"...

**Reply by gevezex** [reply]
Somehow I can't get the sandbox env working. So I will paste the code here:

```vue
<script setup>
</script>

<template>
  <Vueform>
    <SelectElement
      name="selectBrand"
      :native="false"
      label="Select"
      :items="[
        {
          value: 'Mercedes',
          label: 'Mercedes',
        },
        {
          value: 'BMW',
          label: 'BMW',
        },
        {
          value: 'Audi',
          label: 'Audi',
        },
      ]"
    />
    <SelectElement
      name="apiSelect"
      :search="true"
      :native="false"
      label="Select"
      input-type="search"
      autocomplete="off"
      items="https://vpic.nhtsa.dot.gov/api/vehicles/getmodelsformake/{selectBrand}?format=json"
      value-prop="Model_Name"
      label-prop="Model_Name"
      data-key="Results"
      :delay="0"
    />
  </Vueform>
</template>

<style>
</style>
```

**Reply by adamberecz** [reply]
You can pass in a default value for the endpoint param as `{selectBrand|'default'}`. This should work for endpoints that can return an empty response.

A trickier solution for cases when providing a default is not an option:
- turn of `resolveOnLoad` to prevent initial fetch
- set conditions to prevent errors when user starts typing in `apiSelect` first
- and refetch items for `apiSelect` the first time `selectBrand` changes

```vue
<template>
  <Vueform>
    <SelectElement
      name="selectBrand"
      :native="false"
      label="Select"
      :items="[
        {
          value: 'Mercedes',
          label: 'Mercedes',
        },
        {
          value: 'BMW',
          label: 'BMW',
        },
        {
          value: 'Audi',
          label: 'Audi',
        },
      ]"
      @change="handleChange" // refetch apiSelect items
    />
    <SelectElement
      name="apiSelect"
      :search="true"
      :native="false"
      label="Select"
      input-type="search"
      autocomplete="off"
      :resolve-on-load="false" // prevent initial fetch
      items="https://vpic.nhtsa.dot.gov/api/vehicles/getmodelsformake/{selectBrand}?format=json"
      value-prop="Model_Name"
      label-prop="Model_Name"
      data-key="Results"
      :delay="0"
      :conditions="[['selectBrand', '!=', null]]" // prevent user initiated fetch
    />
  </Vueform>
</template>

<script setup>
import { nextTick } from 'vue'

const handleChange = (n, o, el$) => {
  nextTick(() => {
    if (!el$.form$.el$('apiSelect').input.extendedOptions.length) {
      el$.form$.el$('apiSelect').updateItems()
    }
  })
}
</script>
```
